superset/charts/dao.py

```diff
@@ -51,3 +56,25 @@ def bulk_delete(models: Optional[List[Slice]], commit: bool = True) -> None:
             if commit:
                 db.session.rollback()
             raise ex
+
+    @staticmethod
+    def fetch_unique_column_values(column_name: str) -> List:
+        return [
+            val[0]
+            for val in db.session.query(getattr(Slice, column_name)).distinct().all()
+        ]
+
+    @staticmethod
+    def fetch_unique_datasources() -> Optional[List["BaseDatasource"]]:
```

Comment:
This looks fine, but UX note: It might be confusing to only be able to filter by datasources that are used in charts. Using a list of all datasources and handling unused ones with an empty state might be more ideal.

Reply:
That's a good point. I originally implemented the viz_types by checking unique vals in the db, which ended up informing the decision to go with this implementation. Going to see if if there's a way to collect available viz types instead of relying on created ones...